lib/artifact.js

```diff
@@ -156,34 +173,51 @@ module.exports = (bluebird, fs, logger, request, requestOptions, requestPromise,
             options = requestOptions.createOptions(this.authToken);
             options.json = true;
             options.url = this.uri;
+
+            // Set the timeout on the request to a minute. The thought
+            // behind this is just setting it to something sort of long,
+            // in case Shelf is under a large load.
+            options.timeout = 60000;
             logger.debug(`GET request options: ${JSON.stringify(options, null, 4)}`);
 
-            return new Bluebird((resolve) => {
-                request
+            return new Bluebird((resolve, reject) => {
+                return request
                     .get(options)
                     .on("response", (response) => {
                         logger.debug(`In ${this.constructor.name}.downloadToFile() -> request.on("response")`);
                         logger.debug(`Status code: ${response.statusCode}`);
 
-                        response
+                        if (responseHandler.isErrorCode(response.statusCode)) {
+                            return reject(responseHandler.handleErrorResponse(response));
+                        }
+
+                        return response
                             .pipe(downloadLocation)
                             .on("error", (err) => {
                                 logger.debug(`In ${this.constructor.name}.downloadToFile() -> request.on("response").on("error)`);
 
-                                responseHandler.handleErrorResponse(err);
+                                failed = true;
+
+                                return reject(responseHandler.handleErrorResponse(err));
                             })
                             .on("finish", () => {
                                 logger.debug(`In ${this.constructor.name}.downloadToFile() -> request.on("response").on("finish")`);
 
-                                // WriteStream.path can return a Buffer, so let's toString() it
-                                // just in case.
-                                return resolve(downloadLocation.path.toString());
+                                if (!failed) {
+                                    // WriteStream.path can return a Buffer, so let's toString() it
```

Comment:
I got this from the Node.js docs. If you initially pass it a Buffer for the path, it will return a Buffer in the path. I can explain more. Fixed. a221c93